### PR TITLE
Re-add wildcard types and modify checks for acceptable types.

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3810

# What does this Pull Request do?
Re-adds wildcards to the Jersey endpoints otherwise Jersey rejects `Accept` headers that aren't in the list (which are all RDF types and `text/plain` and `text/html`).

Also makes HEAD requests respond the same as a GET request.

# How should this be tested?

Before this PR.
Put up a binary that is not `text/plain` or `text/html` (you can put up plain text and claim it is `application/pdf` though)
ie.
```
> curl -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-type: application/pdf" --data-binary "Some fake binary" http://localhost:8080/rest/test_binary

HTTP/1.1 201 Created
```

Do a HEAD request to see the Content-type is set correctly.
```
> curl -I -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary

HTTP/1.1 200 OK
...
Content-Type: application/pdf
...
```

Now do a HEAD and/or GET request using that mimetype with the Accept header.
```
> curl -I -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary -H"Accept: application/pdf"

HTTP/1.1 406 Not Acceptable
```
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary -H"Accept: application/pdf"

HTTP/1.1 406 Not Acceptable
```

Pull in this PR, rebuild and try getting them again.
```
> curl -I -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary -H"Accept: application/pdf"             

HTTP/1.1 200 OK
```
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary -H"Accept: application/pdf"             

HTTP/1.1 200 OK
```

But it still fails if the type is not compatible.
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary/fcr:metadata -H"Accept: application/pdf"

HTTP/1.1 406 Not Acceptable
```
```
> curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/test_binary -H"Accept: image/tiff"                  

HTTP/1.1 406 Not Acceptable
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers

### CODE REVIEW

**Review by dbernstein — APPROVED** [review]
Nice work @whikloj.  Code looks good and it works as advertised.

**Review by bbpennel — APPROVED** [review]
This looks good to me and is functioning as I would expect for GET and HEAD requests. The only thing I am wondering is if the integration test should verify that `*/*` accept header works as well (which it does in manual testing, but yeah).

**Comment by whikloj** [comment]
I will add that in.

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java [hunk 1]
```diff
@@ -3291,6 +3317,36 @@ public void testResponseContentTypes() throws Exception {
...
+        assertEquals(OK.getStatusCode(), getStatus(httpGet2));
+
+        final HttpHead httpHead = new HttpHead(objectUrl);
+        httpHead.setHeader(ACCEPT, "image/tiff");
+        assertEquals(NOT_ACCEPTABLE.getStatusCode(), getStatus(httpHead));
+
+        final HttpHead httpHead2 = new HttpHead(objectUrl);
+        httpHead2.setHeader(ACCEPT, "application/pdf");
+        assertEquals(OK.getStatusCode(), getStatus(httpHead2));
+    }
```
Comment (bbpennel) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java:3348]: Might be good to test in here what happens when `Accept: */*` is supplied, and no Accept header is provided.

**Comment by bbpennel** [comment]
For RDF sources, should we verify that Accept: */* works as well?

**Comment by whikloj** [comment]
Sorry about that, I read the need for testing `*/*` and then I focused on Head versus Get in the RDF realm. I hope these changes cover everything.

**Pull request merged**